esp-hal/src/time.rs

```diff
@@ -0,0 +1,76 @@
+//! The `time` module offers a way to get the system uptime.
+//!
+//! ### Example
+//! ```no_run
+//! let time = time::current_time();
+//! ```
+#![warn(missing_docs)]
+
+/// Provides time since system start in microseconds precision
+///
+/// The counter won’t measure time in sleep-mode.
+///
+/// The timer will wrap after:
+/// - ESP32 = 36_558 years,
+/// - ESP32-S2 = 7_311 years,
+/// - others = > 7 years
+pub fn current_time() -> fugit::Instant<u64, 1, 1_000_000> {
+    #[cfg(esp32)]
+    let (ticks, div) = {
+        // on ESP32 use LACT
+        let tg0 = unsafe { crate::peripherals::TIMG0::steal() };
+        tg0.lactupdate().write(|w| unsafe { w.update().bits(1) });
```

Comment:
Are there any side effects of using stealing this here? Does it affect the other parts of the driver in anyway? Also I can't see these registers anywhere in the TRM... Am I being blind? :D

Reply:
They are not in the TRM - it's pretty much undocumented (I stumbled upon this by looking into esp-idf to see how they implemented it)

I tested a few things and I don't see anything causing side effects here (I don't see these registers used anywhere else). Unfortunately, there is nothing but the register definition in esp-idf to rely on

Reply:
I found that here: https://github.com/espressif/esp-idf/blob/636ff35b52f10e1a804a3760a5bd94e68f4b1b71/components/esp_timer/src/esp_timer_impl_lac.c

Reply:
Hmm, I wonder how initialization of timg0 affects this, we reset the whole tg subsystem on driver creation right? Could you look into that?

If the LAC timer is available on all chips, maybe for maintainability we should use it always? Just a thought, still happy to use systimer if we choose to do so.

Reply:
Seems LACT is only available on ESP32 and ESP32-S2 - IMHO it's easier to remember there is one chip which is different 🤔 

We don't reset TG on driver creation. From what I see we only configure the clock source (which is fixed to APB on ESP32)

If we would reset TIMG we would also reset WDT and pretty soon see the chip reboot. So I guess it should be fine

Reply:
Thanks for looking into that! Okay, let's stick with systimer for everything else.